client/js/widgets/scoreboard.js

```diff
@@ -0,0 +1,287 @@
+import { $, $a, removeFromDOM, asArray, escapeID } from '../domhelpers.js';
+import { Widget } from './widget.js';
+
+class ScoreBoard extends Widget {
+  constructor(object, surface) {
+    super(object, surface);
+
+    this.addDefaults({
+      movable: true,
+      width: 300,
+      height: 200,
+      layer: -1,
+      typeClasses: 'widget scoreboard',
+      usePlayerColors: true,
+      playersInColumns: true,
+      rounds: null,
+      sortField: 'index',
+      sortAscending: true,
+      scoreProperty: 'score',
+      includeAllSeats: false,
+      showTotals: true,
+      seats: null,
+      showAllRounds: false,
+      roundLabel: 'Round',
+      currentRound: null
+    });
+  }
+
+  applyDeltaToDOM(delta) {
+    if(delta) // Don't call super unless this is a real delta to the scoreboard rather than from a seat
+      super.applyDeltaToDOM(delta);
+    const seats = this.getIncludedSeats();
+    if(seats !== null)
+      this.tableCreate(seats)
+  }
+
+  get(property) {
+    if(property != '_totals')
+      return super.get(property)
+    else if(!this.get('showTotals'))
+      return [];
+    else {
+      // First get total score for each relevant seat
+      const totals = [];
+      let seats = this.getIncludedSeats();
+      if(Array.isArray(seats)) {// Getting seat totals
+        for (const seat of seats) {
+          const score = seat.get(this.get('scoreProperty'));
+          const index = seat.get('index');
+          totals[index] = this.getTotal(asArray(score).map(s => parseFloat(s)));
+          if(isNaN(totals[index]))
+            totals[index] = 0;
+        }
+        return totals
+      } else if (typeof seats == 'object') { // Getting team totals
```

Comment:
This also triggers for `seats === null`.

Reply:
This actually can't happen since the call to `tableCreate` is guarded by seats != null.